File: packages/ui/ios/NativeImagePreviewModule.swift
import ExpoModulesCore

public class NativeImagePreviewModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ZKitNativeImagePreview")

    View(NativeImagePreviewView.self) {
      Events("onIndexChange")

      OnViewDidUpdateProps { (view) in
        view.setupImageView()
      }

      Prop("urls") { (view, urls: [String]?) in
        view.urls = urls
      }

      Prop("items") { (view, items: [[String: Any]]?) in
        view.items = items
      }

      Prop("index") { (view, index: Int?) in
        view.initialIndex = index
      }

      Prop("previewGroupId") { (view, previewGroupId: String?) in
        view.previewGroupId = previewGroupId
      }

      Prop("theme") { (view, theme: Theme?) in
        view.theme = theme ?? .dark
      }
      Prop("closeIconName") { (view, closeIconName: String?) in
        view.closeIconName = closeIconName
      }
      Prop("rightNavItemIconName") { (view, rightNavItemIconName: String) in
        view.rightNavItemIconName = rightNavItemIconName
      }

    }
  }

  func onIndexChange(index: Int) {
    sendEvent("onIndexChange", ["currentIndex": index])
  }
}

File: packages/ui/android/src/main/java/cn/fontree/zkit/ui/preview/NativeImagePreviewModule.kt
package cn.fontree.zkit.ui.preview

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class NativeImagePreviewModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("ZKitNativeImagePreview")

        View(NativeImagePreviewView::class) {
            Events(
                "onIndexChange"
            )
            Prop("theme") { view: NativeImagePreviewView, theme: Theme ->
                view.theme = theme
            }
            Prop("items") { view: NativeImagePreviewView, items: Array<Map<String, Any?>> ->
                view.items = items.toList()
            }
            Prop("urls") { view: NativeImagePreviewView, urls: Array<String> ->
                view.urls = urls
            }
            Prop("index") { view: NativeImagePreviewView, index: Int ->
                view.initialIndex = index
            }
            Prop("previewGroupId") { view: NativeImagePreviewView, previewGroupId: String? ->
                view.previewGroupId = previewGroupId
            }
            Prop("disableHiddenOriginalImage") { view: NativeImagePreviewView, disableHiddenOriginalImage: Boolean ->
                view.disableHiddenOriginalImage = disableHiddenOriginalImage
            }
            Prop("edgeToEdge") { view: NativeImagePreviewView, edgeToEdge: Boolean ->
                view.edgeToEdge = edgeToEdge
            }
            Prop("transitionOffsetY") { view: NativeImagePreviewView, transitionOffsetY: Int? ->
                view.transitionOffsetY = transitionOffsetY
            }
            Prop("transitionOffsetX") { view: NativeImagePreviewView, transitionOffsetX: Int? ->
                view.transitionOffsetX = transitionOffsetX
            }
        }
    }
}
